asmcomp/selectgen.ml

```diff
@@ -395,8 +395,9 @@ method mark_instr = function
 
 (* Default instruction selection for operators *)
 
-method select_allocation words =
-  Ialloc { words; spacetime_index = 0; label_after_call_gc = None; }
+method select_allocation () =
```

Comment:
I don't know much about this piece of code, but it's surprising that words was always 0.  This comes from cd0bd8aa73.  @mshinwell : can you comment?  Was it intentional, and did you plan for possible extensions/changes where this would no longer be true?

Reply:
That's indeed something I figured out yesterday and that I would have reported: instruction selection hasn't happened yet, so words is always zero at this point. The size is computed later, in `emit_expr`.

Anyway, the name is wrongly chosen, since this field actually contain the size in bytes and not in words. Related to this, the assertion ` assert (words <= Config.max_young_wosize);` in `emit_expr` is completely useless (since words in `0` anyway), and checks the wrong thing (since `words` is in bytes and `Config.max_young_wosize` is in words).